### DISCUSSION TITLE
Baumgarte Stabilization Gains in Contact Dynamics "constraint_model.corrector.Kp and Kd"

### DISCUSSION THREAD

**Discussion by Cfather** [Simple-Robotics/aligator#179] [Q&A]
Hi,

I have a quick question on tunning Baumgarte Stabilization Gains in terms of ground contact constraints.
In `talos_walk.py`, the gains are chosen as:
```
cm.corrector.Kp[:] = (0, 0, 100, 0, 0, 0)
cm.corrector.Kd[:] = (50, 50, 50, 50, 50, 50)
```
I wonder why the P gain on the translation force in z direction only is chosen to be 100 while others are 0? And why are Kd non-zero for all directions while Kp are 0? 
Does this effectively stabilize the contact constraints? Are there any tips on tuning these parameters and validating their performance?

I appreciate any suggestions! Thanks.

**Comment by EthanZengg** [comment]
Hello! I encountered the same issue. Have you managed to resolve it?
